vendor/github.com/opencontainers/runc/libcontainer/state_linux.go

```diff
@@ -38,7 +38,8 @@ type containerState interface {
 }
 
 func destroy(c *linuxContainer) error {
-	if !c.config.Namespaces.Contains(configs.NEWPID) {
+	if !c.config.Namespaces.Contains(configs.NEWPID) ||
+		c.config.Namespaces.PathOf(configs.NEWPID) != "" {
```

Comment:
Please do not modify vendor directory directly. The change will be lost next time we update go vendor. If you believe something is broken, please fix it in the original repository (`runc` in this case) and update kata's vendor.conf instead.